lib/jobs/dell-racadm-catalog-job.js

```diff
@@ -0,0 +1,111 @@
+
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = racadmCatalogJobFactory;
+di.annotate(racadmCatalogJobFactory, new di.Provide('Job.Dell.RacadmCatalog'));
+di.annotate(racadmCatalogJobFactory, new di.Inject(
+    'Job.Base',
+    'JobUtils.RacadmTool',
+    'Logger',
+    'Util',
+    'Assert',
+    'Promise',
+    '_',
+    'Services.Waterline',
+    'JobUtils.JobHelpers'
+));
+
+function racadmCatalogJobFactory(
+    BaseJob,
+    racadmTool,
+    Logger,
+    util,
+    assert,
+    Promise,
+    _,
+    waterline,
+    jobHelper
+) {
+    var logger = Logger.initialize(racadmCatalogJobFactory);
+    function RacadmCatalogJob(options, context, taskId) {
+        RacadmCatalogJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = this.context.target;
+        this.action = options.action;
+        this.cifsInfo = {user: options.serverUsername, password: options.serverPassword ,
+            filePath: options.serverFilePath};
+    }
+    util.inherits(RacadmCatalogJob, BaseJob);
+
+    /**
+     * @memberOf RacadmCatalogJob
+     */
+    RacadmCatalogJob.prototype._run = function() {
+        var self = this;
+        waterline.nodes.findByIdentifier(self.nodeId)
+            .then(function (node) {
+                assert.ok(node, 'No node for dell racadm catalog');
+                var obmSetting = _.find(node.obmSettings, { service: 'ipmi-obm-service' });
+                assert.ok(obmSetting, 'No ipmi obmSetting for dell racadm catalog');
+                return obmSetting.config;
+            })
+            .then(jobHelper.revealSecrets)
+            .then(jobHelper.lookupHost)
+            .then(function(obmSetting) {
+                return racadmTool.getConfigCatalog(obmSetting.host, obmSetting.user,
```

Comment:
Suggest still using racadmTool[action] mode in case in future you want to get other catalogs. But I am OK with this job itself.